draft-sharabayko-mops-srt.md

```diff
@@ -2105,380 +2130,210 @@ with a key establishment method.
+In the protocol for the wrap there is a padding, which is a known template, so the responder
+knows from the KM that it has the right KEK to decode the SEK.
+The SEK (generated and transmitted by the initiator) is random, and cannot be known in advance.
+The KEK formula is calculated on both sides, with the difference that the responder gets
+the key length (KLen) from the initiator via the key material (KM).
+It is the initiator who decides on the configured length. The responder obtains it from
+the material sent by the initiator.
+
+The responder returns the same KM message to show that it has the same information as the initiator,
+and that the encoded material will be decrypted. If the responder does not return this status,
+this means that it does not have the SEK. All incoming encrypted packets received by the responder
+will be lost (undecrypted). Even if they are transmitted
+successfully, the receiver will be unable to decrypt them, and so packets will be dropped.
+All data packets coming from responder will be unencrypted.
+
+### KM Refresh {#sec-crypt-km-refresh}
 
 The short lived SEK is regenerated for cryptographic reasons when enough packets have been
-encrypted with it by implementation of either side. One example implementation has default
-value of 2^25, with a pre-announcement period of 4000 packets (i.e. a new key is generated,
-wrapped, and sent at 2^25 minus 4000 packets). Even and odd keys are alternated this way.
-The packets with the earlier key (we will call it key #1, or "odd") will continue to be sent.
+encrypted. The KM refresh period is determined by the implementation.
+The receiver knows which SEK (odd or even) was used to encrypt the packet
+by means of the KK field of the SRT Data Packet ({{data-pkt}}).
+
+There are two variables used to determige the KM Refresh timing:
+
+- KM Refresh Period detemines the number of packets to be sent between two consecutive
+  switch points,
+
+- KM Pre-Announcement Period determines the number of packets before switch point when
+  the new key is announced., and after the switch point to decommission the old key.
+
+The recommended value of the KM Refresh Period is after 2^25 packets encrypted with the same SEK are sent.
+The recommended KM Pre-Announcement Period is 4000 packets (i.e. a new key is generated,
+wrapped, and sent at 2^25 minus 4000 packets).
+
+Even and odd keys are alternated during transmission the following way.
+The packets with the earlier key #1 (let it be the odd key) will continue to be sent.
 The receiver will receive the new key #2 (even), then decrypt and unwrap it.
 The receiver will reply to the sender if it is able to understand.
 Once the sender gets to the 2^25 packet using the odd key (key #1), it will then start to
 send packets with the even key (key #2). It knows that the receiver has what it needs to
 decrypt that even key (key #2). This happens transparently, from one packet to the next.
```

Comment:
```suggestion
Once the sender gets to the 2^25th packet using the odd key (key #1), it will then start to
send packets with the even key (key #2), knowing that the receiver has what it needs to
decrypt them. This happens transparently, from one packet to the next.
```